src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -348,18 +348,42 @@ internal static ImmutableArray<SyntaxReference> GetDeclaringSyntaxReferenceHelpe
             foreach (Location location in locations)
             {
                 // Location may be null. See https://github.com/dotnet/roslyn/issues/28862.
-                if (location == null)
+                if (location == null || !location.IsInSource)
                 {
                     continue;
                 }
-                if (location.IsInSource)
+
+                if (location.SourceSpan.Length != 0)
                 {
-                    SyntaxToken token = (SyntaxToken)location.SourceTree.GetRoot().FindToken(location.SourceSpan.Start);
+                    SyntaxToken token = location.SourceTree.GetRoot().FindToken(location.SourceSpan.Start);
                     if (token.Kind() != SyntaxKind.None)
                     {
                         CSharpSyntaxNode node = token.Parent.FirstAncestorOrSelf<TNode>();
                         if (node != null)
+                        {
                             builder.Add(node.GetReference());
+                        }
+                    }
+                }
+                else
+                {
+                    // Since the location we're interested in can't contain a token, we'll inspect the whole tree,
+                    // pruning away branches that don't contain that location. We'll pick the narrowest node of the type
+                    // we're looking for.
+                    // eg: finding the ParameterSyntax from the empty location of a blank identifier
+                    SyntaxNode parent = location.SourceTree.GetRoot();
+                    SyntaxNode found = null;
+                    foreach (var descendant in parent.DescendantNodesAndSelf(c => c.Location.SourceSpan.Contains(location.SourceSpan)))
```

Comment:
>parent.DescendantNodesAndSelf(c => c.Location.SourceSpan.Contains(location.SourceSpan)) [](start = 47, length = 87)

Could this just be simplified to a call to `Last()`?

Reply:
```cs
parent.DescendentNodes().OfType<TNode>().Last(c => c.Location.SourceSpan.Contains(location.SourceSpan)
```

---
In reply to: [466050264](https://github.com/dotnet/roslyn/pull/46470#discussion_r466050264) [](ancestors = 466050264)

Reply:
If we do that, then we're enumerating all children nodes that don't contain the location of interest. That seems worse.